app/controllers/shared/api.php

```diff
@@ -197,54 +199,89 @@
 
         $authKey = $request->getHeader('x-appwrite-key', '');
 
-        if (!empty($authKey)) { // API Key authentication
+        // API Key authentication
+        if (!empty($authKey)) {
             // Do not allow API key and session to be set at the same time
             if (!$user->isEmpty()) {
                 throw new Exception(Exception::USER_API_KEY_AND_SESSION_SET);
             }
 
-            // Check if given key match project API keys
-            $key = $project->find('secret', $authKey, 'keys');
-            if ($key) {
-                $user = new Document([
-                    '$id' => '',
-                    'status' => true,
-                    'email' => 'app.' . $project->getId() . '@service.' . $request->getHostname(),
-                    'password' => '',
-                    'name' => $project->getAttribute('name', 'Untitled'),
-                ]);
-
-                $role = Auth::USER_ROLE_APPS;
-                $scopes = \array_merge($roles[$role]['scopes'], $key->getAttribute('scopes', []));
-
-                $expire = $key->getAttribute('expire');
-                if (!empty($expire) && $expire < DateTime::formatTz(DateTime::now())) {
-                    throw new Exception(Exception::PROJECT_KEY_EXPIRED);
+            if(str_contains($authKey, '.')) {
```

Comment:
How does the new key looks like? Maybe we should add a more detailed docblock here to explain behavior

Reply:
It's either API key, or JWT token. JWT token consists of 3 (or 4) parts separated by comma. This is the simplest check I could come up with.
Has dot: Is JWT
Doesnt have dot: Is Appwrite API key (hexadecimal)

Reply:
Should we start adding some prefix to keys for future implementations? We might have another one when we want to introduce application to use Appwrite as an OAuth adapter as one example.

Reply:
As discussed on Discord, we now have prefixes `standard_` and `dynamic_` for all API keys.